HPXMLtoOpenStudio/resources/vehicle.rb

```diff
@@ -0,0 +1,218 @@
+
+      discharge_power_act = Model.add_ems_actuator(
+        name: 'battery_discharge_power_act',
+        model_object: ev_elcd,
+        comp_type_and_control: ['Electrical Storage', 'Power Draw Rate']
+      )
+      charge_power_act = Model.add_ems_actuator(
+        name: 'battery_charge_power_act',
+        model_object: ev_elcd,
+        comp_type_and_control: ['Electrical Storage', 'Power Charge Rate']
+      )
+      temp_sensor = Model.add_ems_sensor(
+        model,
+        name: 'site_temp',
+        output_var_or_meter_name: 'Site Outdoor Air Drybulb Temperature',
+        key_name: 'Environment'
+      )
+      discharge_sch_sensor = Model.add_ems_sensor(
+        model,
+        name: 'discharge_sch_sensor',
+        output_var_or_meter_name: 'Schedule Value',
+        key_name: discharging_schedule.name.to_s
+      )
+      charge_sch_sensor = Model.add_ems_sensor(
+        model,
+        name: 'charge_sch_sensor',
+        output_var_or_meter_name: 'Schedule Value',
+        key_name: charging_schedule.name.to_s
+      )
+      soc_sensor = Model.add_ems_sensor(
+        model,
+        name: 'soc_sensor',
+        output_var_or_meter_name: 'Electric Storage Charge Fraction',
+        key_name: elcs.name.to_s
+      )
+
+      ev_discharge_program = Model.add_ems_program(
+        model,
+        name: 'ev_discharge_program'
+      )
+      ev_discharge_program.additionalProperties.setFeature('ObjectType', Constants::ObjectTypeBEVDischargeProgram)
+      unmet_hr_var = OpenStudio::Model::EnergyManagementSystemOutputVariable.new(model, 'unmet_driving_hours')
+      unmet_hr_var.setName('unmet_driving_hours')
+      unmet_hr_var.setTypeOfDataInVariable('Summed')
+      unmet_hr_var.setUpdateFrequency('SystemTimestep')
+      unmet_hr_var.setEMSProgramOrSubroutineName(ev_discharge_program)
+      unmet_hr_var.setUnits('hr')
+
+      # Power adjustment vs ambient temperature curve; derived from most recent data in Figure 9 of https://www.nrel.gov/docs/fy23osti/83916.pdf
+      # This adjustment scales power demand based on ambient temeprature, and encompasses losses due to battery and space conditioning (i.e., discharging losses), as well as charging losses.
+      coefs = [1.412768, -3.910397E-02, 9.408235E-04, 8.971560E-06, -7.699244E-07, 1.265614E-08]
+      power_curve = ''
+      coefs.each_with_index do |coef, i|
+        power_curve += "+(#{coef}*(site_temp_adj^#{i}))"
+      end
+      power_curve = power_curve[1..]
+      ev_discharge_program.addLine("  Set power_mult = #{power_curve}")
+      ev_discharge_program.addLine("  Set site_temp_adj = #{temp_sensor.name}")
+      ev_discharge_program.addLine("  If #{temp_sensor.name} < -17.778")
+      ev_discharge_program.addLine('    Set site_temp_adj = -17.778')
+      ev_discharge_program.addLine("  ElseIf #{temp_sensor.name} > 37.609")
+      ev_discharge_program.addLine('    Set site_temp_adj = 37.609')
```

Comment:
Can you use deg-F here so it's more easily understood where these numbers came from? I think this becomes:

```suggestion
      ev_discharge_program.addLine("  If #{temp_sensor.name} < #{UnitConversions.convert(0, 'F', 'C').round(3)}")
      ev_discharge_program.addLine('    Set site_temp_adj = #{UnitConversions.convert(0, 'F', 'C').round(3)}')
      ev_discharge_program.addLine("  ElseIf #{temp_sensor.name} > #{UnitConversions.convert(100, 'F', 'C').round(3)}")
      ev_discharge_program.addLine('    Set site_temp_adj = #{UnitConversions.convert(100, 'F', 'C').round(3)}')
```